src/vanilla.cpp

```diff
@@ -67,19 +67,18 @@ Rcpp::List europeanOptionEngine(std::string type,
     boost::shared_ptr<QuantLib::StrikedTypePayoff> payoff(new QuantLib::PlainVanillaPayoff(optionType, strike));
     
     if (withDividends) {
-        Rcpp::NumericVector divvalues(discreteDividends), divtimes(discreteDividendsTimeUntil);
-        int n = divvalues.size();
+        std::vector<double> discDividends = Rcpp::as<std::vector<double> >(discreteDividends);
+        std::vector<double> divtimes = Rcpp::as<std::vector<double> >(discreteDividendsTimeUntil);
```

Comment:
I agree this is not needed for this PR. But note that discDividends was filled up inside the loop, so I claim this code should be better. I can take this part out if that hurts your naming sensibility.